AzureMaps/AzureMaps.php

```diff
@@ -0,0 +1,312 @@
+    }
+
+    /**
+     * @return string
+     */
+    private function getSubscriptionKet(): string
+    {
+        return $this->subscriptionKey;
+    }
+
+    /**
+     * @param string $subscriptionKey
+     */
+    private function setSubscriptionKey(string $subscriptionKey)
+    {
+        $this->subscriptionKey = $subscriptionKey;
+    }
+
+    /**
+     * @param string $format
+     */
+    private function setFormat(string $format)
+    {
+        $this->format = $format;
+    }
+
+    /**
+     * @return string
+     */
+    private function getFormat(): string
+    {
+        return $this->format;
+    }
+
+    private function validateResponse(string $content, string $url)
+    {
+        $response = json_decode($content);
+
+        if (!$response) {
+            throw InvalidServerResponse::create($url);
+        }
+
+        if (isset($response->error)) {
+            throw new InvalidServerResponse($response->error->message);
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param $response
+     * @return array
+     */
+    private function formatGeocodeResponse($response): array
+    {
+        return array_map(function ($result) {
+            $builder = new AddressBuilder($this->getName());
+            $coordinates = $result->position;
+            $bounds = $result->viewport;
+
+            $builder->setCoordinates($coordinates->lat, $coordinates->lon);
+            $builder->setBounds(
+                $bounds->btmRightPoint->lat,
+                $bounds->topLeftPoint->lon,
+                $bounds->topLeftPoint->lat,
+                $bounds->btmRightPoint->lon
+            );
+            $builder->setValue('id', $result->id);
+            $builder->setValue('type', $result->type);
+            $builder->setValue('score', $result->score);
+
+            $builder->setStreetName($result->address->streetName ?? null);
+            $builder->setStreetNumber($result->address->streetNumber ?? null);
+            $builder->setCountryCode($result->address->countryCode ?? null);
+            $builder->setCountry($result->address->country ?? null);
+            $builder->setPostalCode($result->address->extendedPostalCode ?? null);
+
+            return $builder->build();
+        }, $response->results);
+    }
+
+    /**
+     * @param $response
+     * @return array
+     */
+    private function formatReverseGeocodeResponse($response): array
+    {
+        return array_filter(array_map(function ($address) {
+            if ($address->position === "0.000000,0.000000") {
+                return null;
+            }
+
+            $builder = new AddressBuilder($this->getName());
+
+            $coordinates = explode(',', $address->position);
+            $latitude = array_shift($coordinates);
+            $longitude = array_shift($coordinates);
+
+            $bounds = $address->address->boundingBox;
+            $southWest = explode(',', $bounds->southWest);
```

Comment:
is there any cases where these can be null?

Reply:
what exactly do you mean by "these" :)